tests/periph_pm/main.c

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (C) 2016-2018 Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Power management peripheral test.
+ *
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "periph/pm.h"
+#ifdef MODULE_PERIPH_RTC
+#include "periph/rtc.h"
+#endif
+#include "pm_layered.h"
+#include "shell.h"
+
+#ifdef MODULE_PM_LAYERED
+static int cmd_block(int argc, char **argv);
+#endif
+static int cmd_lowest(int argc, char **argv);
+static int cmd_off(int argc, char **argv);
+static int cmd_reboot(int argc, char **argv);
+#ifdef MODULE_PM_LAYERED
+static int cmd_set(int argc, char **argv);
+static int cmd_unblock(int argc, char **argv);
+#ifdef MODULE_PERIPH_RTC
+static int cmd_unblock_rtc(int argc, char **argv);
+#endif
+#endif
+
+/**
+ * @brief   List of shell commands for this example.
+ */
+static const shell_command_t shell_commands[] = {
+#ifdef MODULE_PM_LAYERED
+    { "block", "block power mode", cmd_block },
+#endif
+    { "lowest", "enter lowest mode", cmd_lowest },
```

Comment:
Why is this command needed ? if one unblock a power mode with `unblock command`, then the main thread (only thread running) will be blocked (waiting for incoming commands), so the system switch to idle thread that automatically call `pm_set_lowest`.

Reply:
Removed `cmd_lowest`.

Reply:
Out-of-curiosity, do you confirm that it works as expected (in terms of power consumption) ?

Reply:
Yes, I observe a difference when I change the power modes.

![schermafbeelding_2018-03-30_om_10_45_48](https://user-images.githubusercontent.com/815976/38131530-bcf08ee4-3407-11e8-94f1-61aee0e06349.png)

Reply:
really nice!